ports/atmel-samd/supervisor/port.c

```diff
@@ -489,6 +506,11 @@ void port_interrupt_after_ticks(uint32_t ticks) {
         // We'll interrupt sooner with an overflow.
         return;
     }
+#ifdef SAMD21
+    if (hold_interrupt == true) {
```

Comment:
```suggestion
    if (hold_interrupt) {
```

Reply:
I think this will break sleep during PulseIn as-is because we won't have an interrupt to wake us below. I think you'll want to add a check to the `if (!tud_task_event_ready()) {` line to prevent sleep when PulseIn is active.